core-model/src/main/java/com/sequenceiq/cloudbreak/domain/InstanceMetaData.java

```diff
@@ -28,6 +28,10 @@
                         + "WHERE i.instanceGroup.stack.id= :stackId "
                         + "AND i.instanceStatus <> 'TERMINATED' "),
         @NamedQuery(
+                name = "InstanceMetaData.findAllWithTerminatedInStack",
```

Comment:
the name of the finder suggests that the stack is terminated... is that right? if so there is a missing condition i believe